src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidRunCommand.cs

```diff
@@ -205,6 +205,7 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
                 if (instrumentationExitCode != expectedExitCode)
                 {
                     logger.LogError($"Non-success instrumentation exit code: {instrumentationExitCode}, expected: {expectedExitCode}");
+                    return ExitCode.TESTS_FAILED;
```

Comment:
I am not an Android testing expert (I'd ping @akoeplinger  / @steveisok ?) but if this is like regular xunit, I don't think we want to coerce every value to mean "tests failed", just the "1" case from Xunit tests.  

What I'm curious of from the experts is that there could be crash scenarios, for instance, that return some kind of HResult.  Would it be OK just to return the instrumentationExitCode in this case?

Reply:
Another option would be to add an option to XHarness that would just forward the instrumentation exit code and then the person who owns the app can decide how it should behave?

Reply:
> I don't think we want to coerce every value to mean "tests failed", just the "1" case from Xunit tests.

But before this change we returned `GENERAL_FAILURE` in that case, it doesn't matter one test failed or more.

Reply:
@premun it's good idea, but in that case our exit code is the same as the one from instrumentation. We probably should use separate values for them, so the app owner can understand was is from instrumentation or not

Reply:
> > I don't think we want to coerce every value to mean "tests failed", just the "1" case from Xunit tests.
> 
> But before this change we returned `GENERAL_FAILURE` in that case, it doesn't matter one test failed or more.

I definitely don't think general failure describes the scenario of "everything worked as expected except one or more tests failed".  That's quite a specific failure.  To give context about the problem here, especially while standing up new machines but also randomly and when upgrading existing ones, we end up with machines where the device is missing, the device connection times out repeatedly, etc.  Having as many different exit codes for these states as possible, as well as not having a shared exit code that means this non-exceptional circumstance, helps find broken machines faster so DDFUN can offline them.

Reply:
So I think in the case of Android we _can_ detect if things went wrong horribly (crash etc) since then we don't get the `return-code` instrumentation key.

I think it's fine to return `TESTS_FAILED` if we did get a `return-code` but it didn't match what we were expecting, and return `GENERAL_FAILURE` in other cases.

Reply:
Works for me.  We need this change or similar soon because right now it's causing "test failure / test crash" to return the same as "device lost its auth" and making it nearly impossible to broadly catch when newly online hardware is setup wrong or when hardware loses its auth.